### PR TITLE
docs(core): add new example for HostListener

### PR DESCRIPTION
**Pull request by MrJithil**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
This PR will add new example for the HostListner usage.

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #34165

## What is the new behavior?
new example for how listen to global events

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/metadata/directives.ts [hunk 1]
```diff
@@ -872,6 +872,29 @@ export interface HostListener {
...
+ *
+ * ```
+ *
+ * We could also register other DOM event handlers as below.
+ * ``` ts
+ * import { HostListener, Component } from "@angular/core";
+ *
+ * @Component({
+ *   selector: 'app',
+ *   template: `<h1>Hello, You have pressed keys {{counter}} of times! </h1> Youn can press keys to
```
Comment (kapunahelewong) [inline packages/core/src/metadata/directives.ts]: "You"--> "you" (just make lowercase)

**Comment by kapunahelewong** [comment]
Thank you so much for this, @MrJithil! 🌟Just one little note. Will also request @jbogarthyde to review as she does lots of API docs work.

File: packages/core/src/metadata/directives.ts (hunk 1 above)
Comment (jbogarthyde) [reply packages/core/src/metadata/directives.ts]: and another typo, "Youn can press keys" -> "You can ..."
Maybe change to "Press any key to increment the counter."

File: packages/core/src/metadata/directives.ts [hunk 2]
```diff
@@ -872,6 +872,29 @@ export interface HostListener {
  *   template: '<button counting>Increment</button>',
  * })
  * class App {}
+ *
+ * ```
+ *
+ * We could also register other DOM event handlers as below.
```
Comment (jbogarthyde) [inline packages/core/src/metadata/directives.ts]: The following example registers another DOM event handler that listens for key-press events.

**Review by jbogarthyde — CHANGES_REQUESTED** [review]
Suggesting some style changes to match the existing style.

**Review by jbogarthyde — APPROVED** [review]
😊

File: packages/core/src/metadata/directives.ts (hunk 1 above)
Comment (MrJithil) [reply packages/core/src/metadata/directives.ts]: changed to the suggested sentence.

File: packages/core/src/metadata/directives.ts (hunk 2 above)
Comment (MrJithil) [reply packages/core/src/metadata/directives.ts]: Made changes.

**Review by kapunahelewong — APPROVED** [review]
LGTM. Thank you, @MrJithil!

merge-assistance: Global approval for this docs-only change.

**Pull request closed without merging**